vendor/k8s.io/kubernetes/pkg/controller/statefulset/stateful_set.go

```diff
@@ -450,7 +450,11 @@ func (ssc *StatefulSetController) sync(key string) error {
 // syncStatefulSet syncs a tuple of (statefulset, []*v1.Pod).
 func (ssc *StatefulSetController) syncStatefulSet(set *apps.StatefulSet, pods []*v1.Pod) error {
 	glog.V(4).Infof("Syncing StatefulSet %v/%v with %d pods", set.Namespace, set.Name, len(pods))
-	if err := ssc.control.UpdateStatefulSet(set, pods); err != nil {
+	setCopy, err := api.Scheme.DeepCopy(set)
```

Comment:
> I also didn't see anything in the call tree below this that mutated without copying first... I might be missing something, but the fact that the diff was only on the kind and apiVersion fields is odd to me

Suggests that it's getting converted, encoded, or something.  It wasn't obvious to me either.